generation.go

```diff
@@ -0,0 +1,388 @@
+package gen
+
+import (
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+type Generation struct {
+	api *API
+
+	enums     []*Enum
+	functions []*Function
+	structs   []*Struct
+
+	lookupEnum        map[string]*Enum
+	lookupNonTypedefs map[string]string
+	lookupStruct      map[string]*Struct
+}
+
+func newGeneration(a *API, headerFiles []*HeaderFile) *Generation {
+	gen := &Generation{
+		api: a,
+
+		lookupEnum:        map[string]*Enum{},
+		lookupNonTypedefs: map[string]string{},
+		lookupStruct: map[string]*Struct{
+			"cxstring": &Struct{
+				Name:  "cxstring",
+				CName: "CXString",
+			},
+		},
+	}
+	gen.init(headerFiles)
+
+	return gen
+}
+
+func (g *Generation) init(headerFiles []*HeaderFile) {
+	for _, h := range headerFiles {
```

Comment:
I thought about point 1 and 2 yesterday, but is it truly necessary? 
The struct, enums are only defined in one file.
But if a function for said struct or enums is found it is at first added to functions, and only during the generation in *func (g *Generation) gen() error * added to the struct/enum as methods.

Initially only accessors are added to struct/enum methods. That was at least what i found during reading the code, correct me if im wrong